Xamarin.Forms.Xaml/ExpandMarkupsVisitor.cs

```diff
@@ -78,17 +78,28 @@ public void Visit(ListNode node, INode parentNode)
 			if (expression.StartsWith("{}", StringComparison.Ordinal))
 				return new ValueNode(expression.Substring(2), null);
 
-			if (expression[expression.Length - 1] != '}')
-				throw new Exception("Expression must end with '}'");
+			if (expression[expression.Length - 1] != '}') {
+				var ex = new XamlParseException("Expression must end with '}'", xmlLineInfo);
+				if (Context.ExceptionHandler != null) {
+					Context.ExceptionHandler(ex);
+					return null;
+				}
+				throw ex;
+			}
 
-			int len;
-			string match;
-			if (!MarkupExpressionParser.MatchMarkup(out match, expression, out len))
+			if (!MarkupExpressionParser.MatchMarkup(out var match, expression, out var len))
 				throw new Exception();
```

Comment:
Should we also handle this case?

Reply:
@TimBarham we will, but not before one comes up with a test that triggers it. not throwing is easy, but making sure everything else keeps working require a test.

also, if this throws an Exception without message, it's more than probably a safe guard against my own stupidity than a check for invalid markup